src/ring_discord_auth/core.clj

```diff
@@ -24,49 +25,10 @@
            then)
         ~else))))
 
-(defmacro examples
-  "Macro to generate illustrative tests for a function based on input-output pairs.
-
-  `f` is the function to test.
-  `equals` is the function used to compare the expected and actual result.
-  Each example call is a sequence where the first n - 1 items are the inputs and the last item is the expected output."
-  {:style/indent 1}
-  [f equals & example-calls]
-  `(fn []
-     ~@(for [call example-calls
-             :let [in (drop-last call)
-                   out (last call)
-                   out-ev (eval out)]]
-         `(is (~equals (~f ~@in) ~out)))))
-
-(defn hex->bytes
-  "Converts the given string representing a hexadecimal number to a byte array.
-
-  Each byte in the resulting array comes from 2 digits in the string.
-  If the string cannot be converted, returns `nil`"
-  {:test (examples hex->bytes Arrays/equals
-           ["68c252fa74eb9b97" #_=> (byte-array [0x68 0xc2 0x52 0xfa 0x74 0xeb 0x9b 0x97])]
-           ["123" #_=> (byte-array [0x12 0x03])]
-           ["garbage" #_=> nil])}
-  ^bytes [^String hex-str]
-  (let [len (count hex-str)
-
-        result (byte-array (quot (inc len) 2))]
-    (try
-      (doseq [[i hex-part] (map-indexed vector (map (partial apply str) (partition-all 2 hex-str)))]
-        (aset result i (unchecked-byte (Short/parseShort hex-part 16))))
-      result
-      (catch NumberFormatException _ nil))))
-
 (defn read-all-bytes
   "Reads all bytes from either an `InputStream` or a `ByteBuffer`.
-
   If an `InputStream` is provided, it will be consumed, but not closed.
   Returns its result as a *new* byte array."
-  {:test (examples read-all-bytes Arrays/equals
-           [(ByteBuffer/allocate 0) #_=> (byte-array 0)]
-           [(io/input-stream (byte-array [0x12 0xab 0x4f])) #_=> (byte-array [0x12 0xab 0x4f])]
-           [(ByteBuffer/wrap (byte-array [0x67 0x2a 0x4b 0x23 0x5c]) 1 2) #_=> (byte-array [0x2a 0x4b])])}
```

Comment:
I just thought they are very noisy to have in the middle of actual code, instead of then I prefer writing unit tests as examples, but you said something I didn't thought about the libs documentation, I will but then back.
:+1: